src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -360,19 +360,28 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
         private int ComputeHashCode()
         {
             int code = this.OriginalDefinition.GetHashCode();
-            code = Hash.Combine(this.ContainingType, code);
-
-            // Unconstructed method may contain alpha-renamed type parameters while
-            // may still be considered equal, we do not want to give different hashcode to such types.
-            //
-            // Example:
-            //   Having original method A<U>.Goo<V>() we create two _unconstructed_ methods
-            //    A<int>.Goo<V'>
-            //    A<int>.Goo<V">     
-            //  Note that V' and V" are type parameters substituted via alpha-renaming of original V
-            //  These are different objects, but represent the same "type parameter at index 1"
-            //
-            //  In short - we are not interested in the type arguments of unconstructed methods.
+
+            // If the containing type of the of the original definition is the same as our containing type
+            // it's possible that we will compare equal to the original definition under certain conditions 
+            // (e.g, ignoring nullability) and want to retain the same hashcode. As such only make
+            // the containing type part of the hashcode when we know equality isn't possible
+            var containingHashCode = _containingType.GetHashCode();
+            if (containingHashCode != this.OriginalDefinition.ContainingType.GetHashCode())
```

Comment:
>if (containingHashCode != this.OriginalDefinition.ContainingType.GetHashCode()) [](start = 12, length = 79)

I think we should reverse the condition and return what we have. Otherwise, do what we did before. The ```if``` about type arguments below shouldn't be executed if this condition is false.